### PR TITLE
Fix various bugs related to the mouse on Wayland

### PR DESCRIPTION
**Pull request by quantum5**

### CODE REVIEW

File: client/src/main.c [hunk 1]
```diff
@@ -1126,6 +1131,7 @@ static void handleWindowLeave()
 static void handleWindowEnter()
 {
   g_cursor.inWindow = true;
+  g_state.focused   = true;
```
Comment (gnif) [inline client/src/main.c:1134]: Entered is not the same as focused, this is incorrect. Enter means the mouse is over the window, but the window may not have input focus.

Comment (quantum5) [reply client/src/main.c]: Fixed.

**Comment by gnif** [comment]
I think you're trying to fix here a bug we are seeing on X11 also. The trigger is that if the mouse is for any reason moved outside of the window such as toggling full screen while the cursor is in the upper right quadrant of the LG window, the application can not be re-focused and as such clicking breaks.

Also, how will Wayland work if you can't grab the cursor anymore? All movements are based on relative x/y and if the cursor moves outside of the LG client, input will stop.

**Comment by gnif** [comment]
Please see 76ed75f871847111714bc271afb5f780cde45740. This should help with this also and for the most part, make this PR obsolete.

**Comment by quantum5** [comment]
Superseded by #359.

**Pull request closed without merging**